app/src/main/java/com/example/feelgoodinc/SignupActivity.java

```diff
@@ -126,8 +129,17 @@ public void onSuccess(User user) {
                 @Override
                 public void onError(Exception e) {
                     // Registration failed
-                    Toast.makeText(getApplicationContext(),
-                            "Could not sign up", Toast.LENGTH_LONG).show();
+
+                    if (e instanceof IllegalArgumentException && e.getMessage() != null) {
```

Comment:
It feels a bit spaghetti to pass an exception as the way to check if the error is that the password wasn't parsed correctly.

It might be better to make a sepreate interface with a `onInvalidPassword` or create an explicit `InvalidPasswordException` sub-class of Exception.